src/NHibernate/Logging.cs

```diff
@@ -76,7 +76,24 @@ public interface IInternalLogger
 		void WarnFormat(string format, params object[] args);
 	}
 
-	public interface INHibernateLogger
+	// Since v5.1
+	// Required till look alike methods taking object are dropped.
+	[Obsolete("Implement and use NHibernate.INHibernateLogger")]
+	public interface ITransitionalInternaLogger : IInternalLogger
```

Comment:
Ok, I had to add some other dummy methods (but less dummy: they work) with this solution. Since we obsoleting passing an object as a message, and that the instance methods takes precedence other extensions even if extensions have a better matching signature, we need these methods for being able to call something not obsoleted in this case and without using code tricks.